packages/integrations/components/pico/usePicoGTMEvents.js

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react';
+import getLogService from '@irvingjs/services/logService';
+
+const log = getLogService('irving:integrations:pico');
+const events = [
+  'pico.user.registered',
+  'pico.user.authenticated',
+  'pico.payment.paid',
+  'pico.paymentMethod.added',
+];
+
+// Add listeners for Pico events.
+const usePicoGTMEvents = (gtmContainerId) => {
+  /**
+   * Return early if we don't have a containerId
+   * or gtag is not initialized.
+   */
+  if (!window || !window.gtag || !gtmContainerId) {
+    return null;
```

Comment:
Think there's ever a chance this hook will run, return null, then reach the `useEffect` on a subsequent run? Might cause React to yell at us if so

Reply:
Thinking of this section: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Reply:
That is an excellent point!